selfdrive/car/card.py

```diff
@@ -143,25 +144,28 @@ def __init__(self, CI=None, RI=None) -> None:
       self.CP.safetyConfigs = [safety_config]
 
     if self.CP.secOcRequired and not self.params.get_bool("IsReleaseBranch"):
-      # Copy user key if available
-      try:
-        with open("/cache/params/SecOCKey") as f:
-          user_key = f.readline().strip()
-          if len(user_key) == 32:
-            self.params.put("SecOCKey", user_key)
-      except Exception:
-        pass
-
-      secoc_key = self.params.get("SecOCKey", encoding='utf8')
-      if secoc_key is not None:
-        saved_secoc_key = bytes.fromhex(secoc_key.strip())
-        if len(saved_secoc_key) == 16:
-          self.CP.secOcKeyAvailable = True
-          self.CI.CS.secoc_key = saved_secoc_key
-          if controller_available:
-            self.CI.CC.secoc_key = saved_secoc_key
+      tsk: TSK = TSK(self)
+      # Try to retrieve the installed key.
+      key: bytes = tsk.get_installed()
+
+      # If no key was installed, attempt to extract one.
+      if key is None: # TODO: and connected to the car
+        key = tsk.extract()
+
+      # Do we have a valid key?
+      if key is None:
+        if True: # TODO: if connected to the car
+          cloudlog.error("TSK: Failed")
+          # TODO: Tell user to put the car in Not Ready To Drive mode + restart comma
```

Comment:
Please help me find a syntax for showing warning messages on the screen. This warning should stay on the screen until the user presses OK.